### PR TITLE
Add deploy step for bosh-topgun worker

### PR DESCRIPTION
**Pull request by taylorsilva**
fixes https://github.com/concourse/prod/issues/43

This PR adds a `put` step to redeploy our bosh topgun worker.

This `put` step, added to when we redeploy ci, ensures that our bosh
external worker is updated at the same time. Had to add `worker_private_key` and `tsa_host_public_key`, at the pipeline level in vault (`concourse/main/concourse/`)

### CODE REVIEW

File: pipelines/concourse.yml [hunk 1]
```diff
@@ -1104,6 +1104,27 @@ jobs:
     - get: bbr-sdk-release
     - get: prod
     - get: ci
+  - put: bosh-topgun-worker-deployment
```
Comment (pivotal-jamie-klassen) [inline pipelines/concourse.yml]: I think this step should have `tags: [topgun]` based on https://concourse-ci.org/tags-step-modifier.html#tags

File: pipelines/concourse.yml [hunk 2]
```diff
@@ -1838,6 +1859,16 @@ resources:
     client_secret: ((bosh_client.secret))
     deployment: concourse-prod
 
+- name: bosh-topgun-worker-deployment
```
Comment (pivotal-jamie-klassen) [inline pipelines/concourse.yml]: I also think this resource should have `tags: [topgun]` https://concourse-ci.org/resources.html#resource-tags

**Comment by taylorsilva** [comment]
@pivotal-jamie-klassen you raise a great point and I believe you are correct! Added that tag, review again please?

File: pipelines/concourse.yml [hunk 3]
```diff
@@ -1104,6 +1104,28 @@ jobs:
...
+  - put: bosh-topgun-worker-deployment
+    tags: [topgun]
+    params:
+      manifest: cbd/cluster/external-worker.yml
+      stemcells:
+      - gcp-xenial-stemcell/*.tgz
+      releases:
+      - concourse-release/*.tgz
+      - bpm-release/*.tgz
+      ops_files:
```
Comment (pivotal-jamie-klassen) [inline pipelines/concourse.yml]: I suspect that the `bosh-deployment-resource` is gonna expect this to be an array. Can you drop the param completely?

File: pipelines/concourse.yml [hunk 4]
```diff
@@ -1104,6 +1104,28 @@ jobs:
...
+      vars_files:
+      - cbd/versions.yml
```
Comment (pivotal-jamie-klassen) [inline pipelines/concourse.yml]: I think we're gonna want to deploy RCs, so can we maybe drop this and just add the relevant values below? In fact, the default for `concourse_version` at least might be `latest`, which would be what we want in this case.

File: pipelines/concourse.yml (hunk 3 above)
Comment (taylorsilva) [reply pipelines/concourse.yml]: Yup, that was a mistake even having it there.

File: pipelines/concourse.yml (hunk 4 above)
Comment (taylorsilva) [reply pipelines/concourse.yml]: Added an ops-file that switches the versions over to `latest`, `dev-versions.yml`

**Comment by taylorsilva** [comment]
Made the changes requested by @pivotal-jamie-klassen. Now using an ops-file that will allow us to use rc versions

**Pull request merged**